README.md

```diff
@@ -1,26 +1,73 @@
 # fastify-websocket
-
-[![Greenkeeper badge](https://badges.greenkeeper.io/fastify/fastify-websocket.svg)](https://greenkeeper.io/)
+[![js-standard-style](https://img.shields.io/badge/code%20style-standard-brightgreen.svg?style=flat)](http://standardjs.com/)  [![Build Status](https://travis-ci.org/fastify/fastify-websocket.svg?branch=master)](https://travis-ci.org/fastify/fastify-websocket) [![Greenkeeper badge](https://badges.greenkeeper.io/fastify/fastify-websocket.svg)](https://greenkeeper.io/)
 
 WebSocket support for [Fastify](https://github.com/fastify/fastify).
 Built upon [websocket-stream](http://npm.im/websocket-stream).
 
-## Example
+## Install
+
+```
+npm install fastify-websocket --save
+```
+
+## Usage
+
+All you need to do is to add it to you project with `register` and you are done !
+
+### Example
 
 ```js
 'use strict'
 
 const fastify = require('fastify')()
 
-fastify.register(require('fastify-websocket'), { handle })
+const options = {
+  verifyClient: function (info, next) {
+    if (info.req.headers['x-fastify-header'] !== 'fastify is awesome !') {
+      next(false) // the connection is not allowed
+    }
+
+    next(true) // the connection is allowed
+  }
+}
```

Comment:
It would be great to support async/await as well.

Reply:
Do you mean adding an async/await example in the doc ?

With an async `verifyClient` hook function the value returned is always truthy because it returns a Promise.

I think the way I named the callback argument `next` in my example can be misleading. I should change this to something like `done`, `allow` or maybe just use the `verifyClient` hook of `ws` without the callback argument (according to [`ws` doc](https://github.com/websockets/ws/blob/master/doc/ws.md#new-websocketserveroptions-callback) then the return value must be a `boolean` to either accept the client connection or not).

I should change the example to this maybe:
```js
'use strict'

const fastify = require('fastify')()

const options = {
  // We call the `verifyClient` hook
  verifyClient: (info) => {
    // Make your verifications here e.g. :
    if (info.req.headers['x-fastify-header'] !== 'fastify is awesome !') {
      return false // the connection is not allowed
    } else {
      return true // the connection is allowed
    }
  }
}

fastify.register(require('fastify-websocket'), {
  handle,
  options
})

function handle (conn) {
  conn.pipe(conn) // creates an echo server
}

fastify.listen(3000, (err) => {
  if (err) {
    fastify.log.error(err);
    process.exit(1);
  }
})
```

WDYT ? Sorry if I didn't understood correctly ^^

Reply:
I didn't understand that `verifyClient` was provided by the `ws` module. This is ok.